GunShipAlpha/src/Enemy.java

```diff
@@ -1,24 +1,64 @@
 import java.awt.*;
+import java.awt.geom.Point2D;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
 
 class Enemy {
     private int x, y;
     private final int targetX, targetY;
     private int health = 2;
+    private Timer firingTimer;
     public Enemy(int x, int y, int targetX, int targetY) {
         this.x = x;
         this.y = y;
         this.targetX = targetX;
         this.targetY = targetY;
+        scheduleFiring();
     }
+    private void scheduleFiring() {
+        if (firingTimer != null) {
+            firingTimer.cancel();
+        }
 
+        firingTimer = new Timer();
+
+        TimerTask firingTask = new TimerTask() {
+            @Override
+            public void run() {
+
+                Projectile projectile = fireAtPlayer();
+                Game.gs.addProjectile(projectile);
+                scheduleFiring();
+            }
+        };
+        Random random = new Random();
+        long delay = 0 + random.nextInt(3000);
+        firingTimer.schedule(firingTask, delay);
+    }
     public void moveTowardTarget() {
+        if (hasEnemyReachedTarget()) return;
         if (x < targetX) x++;
         if (x > targetX) x--;
         if (y < targetY) y++;
         if (y > targetY) y--;
     }
+
+    private boolean hasEnemyReachedTarget() {
+        if(x == targetX && y == targetY ) {
+            health--;
```

Comment:
health -= 2?

Reply:
also looks like this should use removeHealth?

Reply:
I don't know why I had health-- twice instead of health - = 2 lmao, it doesnt use remove health because it needs to kill the enemy fully instead of taking one health off